libevm/JitVM.cpp

```diff
@@ -76,41 +76,51 @@ void queryState(
 	}
 }
 
-void updateState(
+void setStorage(
 	evm_env* _opaqueEnv,
-	evm_update_key _key,
 	evm_uint160be const* _addr,
-	evm_variant const* _arg1,
-	evm_variant const* _arg2
+	evm_uint256be const* _key,
+	evm_uint256be const* _value
 ) noexcept
 {
 	(void) _addr;
 	auto &env = reinterpret_cast<ExtVMFace&>(*_opaqueEnv);
 	assert(fromEvmC(*_addr) == env.myAddress);
-	switch (_key)
-	{
-	case EVM_SSTORE:
-	{
-		u256 index = fromEvmC(_arg1->uint256be);
-		u256 value = fromEvmC(_arg2->uint256be);
-		if (value == 0 && env.store(index) != 0)                   // If delete
-			env.sub.refunds += env.evmSchedule().sstoreRefundGas;  // Increase refund counter
+	u256 index = fromEvmC(*_key);
+	u256 value = fromEvmC(*_value);
+	if (value == 0 && env.store(index) != 0)                   // If delete
+		env.sub.refunds += env.evmSchedule().sstoreRefundGas;  // Increase refund counter
 
-		env.setStore(index, value);    // Interface uses native endianness
-		break;
-	}
-	case EVM_LOG:
-	{
-		size_t numTopics = _arg2->data_size / sizeof(h256);
-		h256 const* pTopics = reinterpret_cast<h256 const*>(_arg2->data);
-		env.log({pTopics, pTopics + numTopics}, {_arg1->data, _arg1->data_size});
-		break;
-	}
-	case EVM_SELFDESTRUCT:
-		// Register selfdestruction beneficiary.
-		env.suicide(fromEvmC(_arg1->address));
-		break;
-	}
+	env.setStore(index, value);    // Interface uses native endianness
+}
+
+void selfdestruct(
+	evm_env* _opaqueEnv,
+	evm_uint160be const* _addr,
+	evm_uint160be const* _beneficiary
+) noexcept
+{
+	(void) _addr;
+	auto &env = reinterpret_cast<ExtVMFace&>(*_opaqueEnv);
+	assert(fromEvmC(*_addr) == env.myAddress);
+	env.suicide(fromEvmC(*_beneficiary));
+}
+
+
+void log(
+	evm_env* _opaqueEnv,
+	evm_uint160be const* _addr,
+	uint8_t const* _data,
+	size_t _dataSize,
+	evm_uint256be const _topics[],
+	size_t _numTopics
+) noexcept
+{
+	(void) _addr;
+	auto &env = reinterpret_cast<ExtVMFace&>(*_opaqueEnv);
+	assert(fromEvmC(*_addr) == env.myAddress);
+	h256 const* pTopics = reinterpret_cast<h256 const*>(_topics);
+	env.log({pTopics, pTopics + _numTopics}, {_data, _dataSize});
```

Comment:
This is little confusing, not clear what exactly is being passed to `log`, I'd probable name the constructed types explicitly, like
`env.log(h256s{pTopics, pTopics + _numTopics}, bytesConstRef{_data, _dataSize});`